src/main/java/com/codeborne/selenide/collections/ContainTexts.java

```diff
@@ -45,14 +45,16 @@ public void fail(CollectionSource collection,
                    @Nullable List<WebElement> elements,
                    @Nullable Exception lastError,
                    long timeoutMs) {
+    List<String> actualTexts = ElementsCollection.texts(elements);
```

Comment:
what about to move this logic under `-else-` ?

Reply:
Good point. 
Initially, I had to put this line first because it was `actualTexts` also used in one of the `else if` branches (which is gone now).

It's fixed now, moved to `else` branch.